javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -162,6 +162,27 @@
     }
 
     /**
+     * Chooses an enum value from all the enum constants defined in the enumerated type.
+     * @param clazz Enum class
+     * @return A new enum generator
+     */
+    static <E extends Enum<E>> Gen<E> choose(Class<E> clazz) {
+        return random -> Gen.choose(clazz.getEnumConstants()).apply(random);
+    }
+
+    /**
+     * Chooses a value from all values in the array.
+     * @param values array with the values to choose from
+     * @return A new enum generator
+     */
+    static <E> Gen<E> choose(E[] values) {
```

Comment:
Done! Happy to contribute.  By the way, since I'm proposing support for arrays, I might as well add for other collections. Would that be ok @danieldietrich ?